components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -419,6 +420,15 @@ protected void concludeFlow(HttpServletRequest request, HttpServletResponse resp
                         log.error("Updating session meta data failed.", e);
                     }
                 }
+                /*
+                 * In the default configuration, the expiry time of the commonAuthCookie is fixed when rememberMe
+                 * option is selected. With this config, the expiry time will increase at every authentication.
+                 */
+                if (sessionContext.isRememberMe() &&
+                        Boolean.parseBoolean(IdentityUtil.getProperty(EXTEND_REMEMBER_ME_SESSION_ON_AUTH))) {
+                    context.setRememberMe(sessionContext.isRememberMe());
+                    setAuthCookie(request, response, context, commonAuthCookie, applicationTenantDomain);
```

Comment:
From how much is the expiry time increased?
is it by `SessionIdleTimeout`?

Reply:
No. In the resident IDP config we specify the remember me timeout. So for each auth, this expiry time will be extended for that specified time.